spec/porky_lib/unencrypted/file_service_spec.rb

```diff
@@ -0,0 +1,203 @@
+
+  it 'read plaintext data from S3 with directory' do
+    dir_name = 'directory1/dirA'
+    file_key = file_service.write(plaintext_data, bucket_name, directory: dir_name)
+    expect(file_key).to include(dir_name)
+
+    plaintext = file_service.read(bucket_name, file_key)
+    expect(plaintext_data).to eq(plaintext)
+  end
+
+  it 'read large plaintext data from S3' do
+    stub_large_file
+    file_key = file_service.write(plaintext_data, bucket_name)
+
+    plaintext, = file_service.read(bucket_name, file_key)
+    expect(File.read("spec#{File::SEPARATOR}porky_lib#{File::SEPARATOR}data#{File::SEPARATOR}large_plaintext")).to eq(plaintext)
+  end
+
+  it 'read plaintext data too large from S3' do
+    stub_large_file
+    file_key = file_service.write(plaintext_data, bucket_name)
+
+    PorkyLib::Config.configure(max_file_size: 10 * 1024)
+    expect do
+      file_service.read(bucket_name, file_key)
+    end.to raise_exception(PorkyLib::Unencrypted::FileService::FileSizeTooLargeError)
+  end
+
+  it 'attempt to write with file nil raises FileServiceError' do
+    expect do
+      file_service.write(nil, bucket_name)
+    end.to raise_exception(PorkyLib::Unencrypted::FileService::FileServiceError)
+  end
+
+  it 'attempt to write with bucket name nil raises FileServiceError' do
+    expect do
+      file_service.write(plaintext_data, nil)
+    end.to raise_exception(PorkyLib::Unencrypted::FileService::FileServiceError)
+  end
+
+  it 'attempt to write to bucket without permission raises FileServiceError' do
+    Aws.config[:s3].delete(:stub_responses)
+    Aws.config[:s3] = {
+      stub_responses: {
+        put_object: 'Forbidden'
+      }
+    }
+    expect do
+      file_service.write(plaintext_data, bucket_name)
+    end.to raise_exception(PorkyLib::Unencrypted::FileService::FileServiceError)
+  end
+
+  it 'attempt to write to bucket that does not exist raises FileServiceError' do
+    Aws.config[:s3].delete(:stub_responses)
+    Aws.config[:s3] = {
+      stub_responses: {
+        put_object: 'NotFound'
+      }
+    }
+    expect do
+      file_service.write(plaintext_data, bucket_name)
+    end.to raise_exception(PorkyLib::Unencrypted::FileService::FileServiceError)
+  end
+
+  it 'attempt to read from bucket without permission raises FileServiceError' do
+    Aws.config[:s3].delete(:stub_responses)
+    Aws.config[:s3] = {
+      stub_responses: {
+        get_object: 'Forbidden'
+      }
+    }
+    expect do
+      file_service.read(bucket_name, default_key_id)
+    end.to raise_exception(PorkyLib::Unencrypted::FileService::FileServiceError)
+  end
+
+  it 'attempt to read from bucket does not exist raises FileServiceError' do
+    Aws.config[:s3].delete(:stub_responses)
+    Aws.config[:s3] = {
+      stub_responses: {
+        get_object: 'NotFound'
+      }
+    }
+    expect do
+      file_service.read(bucket_name, default_key_id)
+    end.to raise_exception(PorkyLib::Unencrypted::FileService::FileServiceError)
+  end
+end
```

Comment:
Maybe add `describe` to encapsulate tests
`describe '#read' do` `describe '#write' do`